up_client_socket/rust/utransport-socket/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+        }
+    };
+       let mut transport_socket = UTransportSocket {
+        socket_sync:socket_clone,
+        listener_map: listener_map.clone(),
+    };
+        match transport_socket.socket_init(){
+            Ok(_) => {
+                // The function call succeeded
+                debug!("socket trasport initilized successfully");
+            }
+            Err(err) => {
+                // The function call failed with an error
+                error!("socket trasport initilized failed: {}", err);
+                return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in cloning socket_sync"));
+            }
+        }
+        debug!("UTransportSocket initilization successful");
+       
+        Ok( UTransportSocket {
+             socket_sync,
+             listener_map,
+         })
+     }
+ 
+     /// # Panics
+     ///
+     /// Will panic if failed to parse message
+      fn socket_init(&mut self) -> Result<(), UStatus>{
+        let socket_clone = match self.socket_sync.try_clone() {
+            Ok(socket) => socket,
+            Err(err) => {
+          
+                error!("Issue in cloning sync socket: {}", err);
+                return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in cloning socket_sync"));
+            }
+        };
+    
+    let listener_map_clone = self.listener_map.clone();
+    
+    let mut self_copy = Self {
+        socket_sync: socket_clone,
+        listener_map: listener_map_clone,
+    };
+    task::spawn(async move { self_copy.dispatcher_listener().await });
+   
+    
+    Ok(())
+}
+    async  fn dispatcher_listener(&mut self) {
+        debug!("started listener for dispatcher");
+         loop {
+             // Receive data from the socket
+             let mut buffer: [u8; BYTES_MSG_LENGTH] = [0; BYTES_MSG_LENGTH];
+             let bytes_read = match self.socket_sync.read(&mut buffer) {
+                 Ok(bytes_read) => bytes_read,
+                 Err(e) => {
+                     debug!("Socket error: {}", e);
+                     break;
+                 }
+             };
+ 
+             // Check if no data is received
+             if bytes_read == 0 {
+                 continue;
+             }
+ 
+             let umessage_result = UMessage::parse_from_bytes(&buffer[..bytes_read]);
+             let umessage = match umessage_result {
+                 Ok(umsg) => umsg,
+                 Err(err) => {
+                     error!("Failed to parse message: {}", err);
+                     continue;
+                 }
+                };
+             
+             match umessage
+                 .attributes
+                 .type_
+                 .enum_value_or(UMessageType::UMESSAGE_TYPE_UNSPECIFIED)
+             {
+                 UMessageType::UMESSAGE_TYPE_PUBLISH => {
+                     debug!("calling handle publish....");
+                     match self.check_all_listeners(&umessage.attributes.source.clone(), umessage){
```

Comment:
Since the value of Ok() is not being used anyway, let's try for this:
```rust
if let Err(err) = self.check_all_listeners(&umessage.attributes.source.clone(), umessage) {
  error!("Error checking listeners: {}", err);
  continue;
} else {
  debug!("All listeners checked successfully");
}
```